### ISSUE TITLE
DragonflyBSD latest git fails to compile

### ISSUE THREAD

**Issue by stormchaser3000** [minetest/minetest#5937]
<pre>
/home/stormchaser3000/minetest/src/porting.cpp: In function 'bool porting::getCurrentExecPath(char*, size_t)':
/home/stormchaser3000/minetest/src/porting.cpp:297:11: error: 'CTL_KERN' was not declared in this scope
  mib[0] = CTL_KERN;
           ^
/home/stormchaser3000/minetest/src/porting.cpp:298:11: error: 'KERN_PROC' was not declared in this scope
  mib[1] = KERN_PROC;
           ^
/home/stormchaser3000/minetest/src/porting.cpp:299:11: error: 'KERN_PROC_PATHNAME' was not declared in this scope
  mib[2] = KERN_PROC_PATHNAME;
           ^
/home/stormchaser3000/minetest/src/porting.cpp:302:39: error: 'sysctl' was not declared in this scope
  if (sysctl(mib, 4, buf, &len, NULL, 0) == -1)
                                       ^
gmake[2]: *** [src/CMakeFiles/minetest.dir/build.make:4743: src/CMakeFiles/minetest.dir/porting.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[ 89%] Building CXX object src/CMakeFiles/minetest.dir/rollback.cpp.o
gmake[1]: *** [CMakeFiles/Makefile2:199: src/CMakeFiles/minetest.dir/all] Error 2
gmake: *** [Makefile:152: all] Error 2
</pre>

**Comment by sfan5** [comment]
@nerzhul

**Comment by nerzhul** [comment]
A DragonflyBSD user wow, it's rare, better to use FreeBSD haha.

Function is bool getCurrentExecPath(char *buf, size_t len) which is not affected by any C++11 change @sfan5  :) it seems there are missing includes in original code from 2014.

Can you try on porting.cpp L28 to replace #if defined(__FreeBSD__) with

```
#if defined(__FreeBSD__)  || defined(__NetBSD__) || defined(__DragonFly__)
```

I think it's just mismatch between inclusion and function usage :)

If it work i will push the fix to master this evening.

Thanks

**Comment by stormchaser3000** [comment]
after the change it compiled just fine.

thanks

**Comment by SmallJoker** [comment]
Fixed in 6c55874
